packages/asc-ui/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { ChevronLeft, ChevronRight } from '@datapunt/asc-assets'
+import PaginationStyle, {
+  Props,
+  PageNumber,
+  PreviousButton,
+  NextButton,
+} from './PaginationStyle'
+
+const Pagination: React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
+  page = 1,
+  collectionSize,
+  pageSize = 10,
+  onPageChange,
+  ...otherProps
+}) => {
+  const [currentPage, setCurrentPage] = React.useState<number>(page)
+  const totalPages = Math.ceil(collectionSize / pageSize)
+  const pageCluster = [currentPage]
+  if (currentPage > 2) {
+    pageCluster.unshift(currentPage - 1)
+  }
+  if (totalPages <= 5 || currentPage < totalPages - 1) {
+    pageCluster.push(currentPage + 1)
+  }
+
+  React.useEffect(() => {
+    if (onPageChange !== undefined) onPageChange(currentPage)
+  }, [currentPage, onPageChange])
+
+  const handlePrevious = () => {
+    setCurrentPage(currentPage - 1)
+  }
+
+  const handleNext = () => {
+    setCurrentPage(currentPage + 1)
+  }
+
+  const renderButton = (pageNumber: number) => {
+    const isCurrent = pageNumber === currentPage
+    return (
+      <PageNumber
+        aria-label={
+          isCurrent ? `Pagina ${pageNumber}` : `Ga naar pagina ${pageNumber}`
+        }
+        aria-current={isCurrent}
+        data-testid={`btn-page-${pageNumber}`}
+        onClick={(evt) => {
+          // Remove focus from button after click
+          evt.currentTarget.blur()
```

Comment:
why do we need to blur after click? I think that's unintended behaviour or am I missing something?

Reply:
No longer necessary after getting rid of the yellow focused state